libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -22,127 +62,134 @@ def store() -> SQLServer_VectorStore:
     assert len(result) == len(texts)
 
 
-def test_add_document_with_sqlserver(store: SQLServer_VectorStore) -> None:
+def test_add_document_with_sqlserver(
+    store: SQLServer_VectorStore,
+    docs: List[Document],
+) -> None:
     """Test that when add_document function is used, it integrates well
     with the add_text function in SQLServer Vector Store."""
-    docs = [
-        Document(
-            page_content="rabbit",
-            metadata={"color": "black", "type": "pet", "length": 6},
-        ),
-        Document(
-            page_content="cherry",
-            metadata={"color": "red", "type": "fruit", "length": 6},
-        ),
-        Document(
-            page_content="hamster",
-            metadata={"color": "brown", "type": "pet", "length": 7},
-        ),
-        Document(
-            page_content="cat", metadata={"color": "black", "type": "pet", "length": 3}
-        ),
-        Document(
-            page_content="elderberry",
-            metadata={"color": "blue", "type": "fruit", "length": 10},
-        ),
-    ]
     result = store.add_documents(docs)
     assert len(result) == len(docs)
 
 
 def test_that_a_document_entry_without_metadata_will_be_added_to_vectorstore(
     store: SQLServer_VectorStore,
+    docs: List[Document],
 ) -> None:
-    docs = [
-        Document(
-            page_content="rabbit",
-            metadata={"color": "black", "type": "pet", "length": 6},
-        ),
-        Document(
-            page_content="cherry",
-        ),
-        Document(
-            page_content="hamster",
-            metadata={"color": "brown", "type": "pet", "length": 7},
-        ),
-        Document(page_content="cat"),
-        Document(
-            page_content="elderberry",
-            metadata={"color": "blue", "type": "fruit", "length": 10},
-        ),
-    ]
-    result = store.add_documents(docs)
-    assert len(result) == len(docs)
+    """Test that you can add a document that has no metadata into the vectorstore."""
+    documents = docs[:-1]
+    documents.append(Document(page_content="elderberry"))
+    result = store.add_documents(documents)
+    assert len(result) == len(documents)
+
+
+def test_that_drop_deletes_vector_store(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+) -> None:
+    """Test that when drop is called, vector store is deleted
+    and a call to add_text raises an exception.
+    """
+    store.drop()
+    with pytest.raises(DBAPIError):
+        store.add_texts(texts)
+
+
+def test_that_similarity_search_returns_expected_no_of_documents(
+    store: SQLServer_VectorStore,
+    texts: List[str],
+) -> None:
+    """Test that the amount of documents returned when similarity search
+    is called is the same as the number of documents requested."""
+    store.add_texts(texts)
+    result = store.similarity_search(query="Good review", k=3)
+    assert len(result) == 3
+
+
+def test_that_similarity_search_returns_results_with_scores_sorted_in_ascending_order(
```

Comment:
We can add more unit tests to get a complete code coverage, because these are all separate APIS's, that can be called individually:

1. Verify similarity_search_by_vector
- returns intended number of results.
- returns are sorted
- for exception cases.

2. Verify similarity_search_with_score 
- returns the score along with the results.
- returns intended number of results.
- Checks for exception cases.

3. Verify similarity_search_by_vector_with_score  
- returns intended number of results along with scores.
- returns are sorted
- for exception cases.

4. similarity_search
- returns are sorted
- for exception cases.

Might be lot of new tests, but these would ensure all the edge cases are covered.

Reply:
Yes, there are more tests we can have from this implementation. The case is all the APIs implemented here have the same base function `_search_store` their differences are in the different helper functions that were created to format results. Tests for those helper functions are actual unit tests, we intend to move this file to the integration test folder instead. That said, the unit tests for those helper functions are available but I'll want it to be in a different PR solely for moving the tests to the appropriate location and including mock tests.

What do you think?

Reply:
Yes, this makes sense. It is better to have a separate PR for the changes related to migrating these tests. Let's discuss the code coverage with tests after we do the migration, if needed.